### PR TITLE
Refactor installer significantly

### LINKED ISSUE #11: Should document how to remove a package
Seems like the installer can't do this right now. Seems like it would be useful to give it the ability...

### LINKED ISSUE #27: Make the upgrade path easier
As we get more and more packages, knowing to upgrade them all is more problematic. Need a simple way that ideally 'just works' and doesn't cause users to miss updates.

First thought was to inspect the local ipk/pip cache and download new versions of those, then on install do the same.

One thought is that pyfrc should probably check all robotpy installed versions when deploying to the robot? Make sure local matches remote, etc.

Or, what about creating a dependency file of some sort, and operating on that? Then users can check it in, and we could sync it both locally and on the remote.

### LINKED ISSUE #40: Move caches to single directory
It's a bit obnoxious to have three different files trash your desktop.

### LINKED ISSUE #41: download-pip can take forever
User reported downloading robotpy-wpilib-utilities and it took forever because pip was trying to build wheels

### LINKED ISSUE #44: Make command error detection not suck
Previously it was a giant bash command because of limitations in the prior ssh. We should segment it so it's easier to diagnose issues.

### LINKED ISSUE #49: Distribute roboRIO wheels instead of ipks
The more I think about this, the better it sounds. Would solve #41 , make CI cross-compiles significantly easier to accomplish, and probably provide a more consistent user experience if done right.

The only real catch here is that we would need a way to know that a binary opkg requirement was needed for some particular package -- for example, cscore requires OpenCV, and scipy requires fortran/blas. I imagine the best way to do this is to publish some list (similar to the way Packages works) that lets robotpy-installer know about these dependencies and let it handle it automatically.

* [x] https://www.tortall.net/~robotpy/wheels/2020/roborio/ contains wheels for most of our binary wheels.
* [x] Creat a wheel building repository via the crossenv container, for things like scipy et al
* [ ] Upgrade robotpy-installer to download the wheels via tortall
* [ ] Remove existing opkg management features and hide them behind a different command
* [ ] Add migration code to robotpy-installer to remove existing ipks and use wheels
* [ ] Update documentation sites

### PR DESCRIPTION
**Pull request by virtuald**
Building on the work started last year, this is a significant set of improvements that should make the installer easier to use, more robust against errors, and easier to add new features to it.

* Fixes #11
* Fixes #27
* Fixes #40
* Fixes #41
* Fixes #44
* Fixes #49
* Fixes #50

### CODE REVIEW

File: robotpy_installer/sshcontroller.py [hunk 1]
```diff
@@ -19,10 +21,21 @@ def missing_host_key(self, client, hostname, key):
...
     """
-        Use this to execute commands on a roboRIO in a
-        cross platform manner
+    Use this to execute commands on a roboRIO in a cross platform manner
+
+    ::
+
+        with SshController(hostname, username, password) as controller:
+            controller.check_call()
+            contro
```
[the remark this answers is not in the document]
Comment (virtuald) [reply robotpy_installer/sshcontroller.py:37]: Nope that's definitely how you use it

File: robotpy_installer/installer.py [hunk 2]
```diff
@@ -1,341 +1,381 @@
...
+    remove_legacy_components(ssh)
 
-        if not options.no_tools:
-            options.packages.append("robotpy-wpilib-utilities")
+    #
+    # Ensure that pip is installed
+    #
 
-        return options
+    if pip_check:
```
[the remark this answers is not in the document]
Comment (virtuald) [reply robotpy_installer/installer.py:177]: pip_check is only set to True for pip operations. If you're trying to install python it won't check for pip.

[the remark this answers is not in the document]
Comment (auscompgeek) [reply robotpy_installer/installer.py:177]: That's my point though, if someone comes along and tries to do `robotpy-installer install robotpy` on top of a 2020 installation, what happens?

[the remark this answers is not in the document]
Comment (virtuald) [reply robotpy_installer/installer.py:177]: It calls remove_legacy_components first

[the remark this answers is not in the document]
Comment (auscompgeek) [reply robotpy_installer/installer.py:177]: Yeah, but then the subsequent message they'd get feels like gaslighting. I can imagine people saying things like:

> I know Python isn't installed, you literally just uninstalled it!

But up to you I guess.

**Pull request merged**